src/sas/qtgui/Plotting/Plotter.py

```diff
@@ -331,46 +331,101 @@ def has_nonempty_plots(self) -> bool:
         return False
 
     def _plot_bounds(self, offset=0.05) -> tuple[tuple[float, float], tuple[float, float]]:
-        """ Get the appropriate bounds for the plots
+        """
+        Get the appropriate bounds for the plots. This is a workaround for a
+        longstanding but edge case bug in matplotlib's autoscale. Basically,
+        data whose range is tiny on a log scale (e.g. 0.9 to 1.1 on a log
+        scale).
+
+        This should be removed when matplotlib fixes the bug. In the meantime
+        this private method ensures that all plots are scaled such that there
+        is a small "white space" between the data extremes and the plot edges.
+
+        In order to achieve this, the min and max of all the data on the plot
+        are computed, taking into account that will include the top, or bottom,
+        of any error bars present with the data. For log axes data that first
+        requires removing any negative values from the min max computation.
 
-        :param offset: add a small fraction of the absolute value of each end to each end
-        :returns:
+        The min and max values for each scale are then converted from data
+        coordinates to axes coordinates before computing the range of the data
+        (max-min). A fraction of that range (default is 5%) is then added and
+        subtracted from the smallest (min) and largest (max) values of x and y,
+        still in axes space. This ensures that regardless of the plot scaling
+        10% of the plot space (5% on either end) will be empty.
+
+        Finally, those adjusted min and max values are converted back to data
+        coordinates to be used in setting the plot bounds in the usual fashion
+
+        :param offset: The fraction of the absolute value of the full
+         data range to add to each end of the data range
+        :returns: lower right (xmin,ymin) and upper left (xmax,ymax) corners of
+         where the plot (figure) bounds should be set.
         """
 
         x_min, x_max = np.inf, -np.inf
         y_min, y_max = np.inf, -np.inf
 
+        # First let's find the smallest xmin and ymin and largest xmax and ymax
+        # in all the data sets. if the data have error bars then we need to
+        # keep track of the data poing + (or -) the error bar height to make
+        # sure the error bars stay within the plot.
         for key in self.plot_dict:
 
             plot_data = self.plot_dict[key].view
 
+            # NOTE: we need to trim any negative values if the scale of the
+            # particular axis is log scale before looking for the minimum
+
+            # First the x axis
+            plot_data.x = [i for i in plot_data.x if i > 0] if self.ax.xaxis.get_scale() == "log" else plot_data.x
```

Comment:
I tested a data set that included negative x and y values, and when using log scale the negative values had already been removed at this point, but I guess an extra check doesn't hurt.

Reply:
Interesting ... While I never tested x, the y axis problem we were seeing was due to negative values coming through? Perhaps it was to do only with the error bars? I can try checking that I guess